nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/PutKudu.java

```diff
@@ -0,0 +1,135 @@
+@Tags({"put", "database", "NoSQL", "kudu", "HDFS"})
+@CapabilityDescription("Reads records from an incoming FlowFile using the provided Record Reader, and writes those records " +
+        "to the specified Kudu's table. The schema for the table must be provided in the processor properties or from your source." +
+        " If any error occurs while reading records from the input, or writing records to Kudu, the FlowFile will be routed to failure")
+
+public class PutKudu extends AbstractKudu {
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(KUDU_MASTERS);
+        properties.add(TABLE_NAME);
+        properties.add(SKIP_HEAD_LINE);
+        properties.add(RECORD_READER);
+        properties.add(INSERT_OPERATION);
+
+        return properties;
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        final Set<Relationship> rels = new HashSet<>();
+        rels.add(REL_SUCCESS);
+        rels.add(REL_FAILURE);
+        return rels;
+    }
+
+    @Override
+    protected Upsert upsertRecordToKudu(KuduTable kuduTable, Record record, List<String> fieldNames) throws IllegalStateException, Exception {
+        Upsert upsert = kuduTable.newUpsert();
+        this.insert(kuduTable, upsert, record, fieldNames);
+        return upsert;
+    }
+
+    @Override
+    protected Insert insertRecordToKudu(KuduTable kuduTable, Record record, List<String> fieldNames) throws IllegalStateException, Exception {
+        Insert insert = kuduTable.newInsert();
+        this.insert(kuduTable, insert, record, fieldNames);
+        return insert;
+    }
+
+    private void insert(KuduTable kuduTable, Operation operation, Record record, List<String> fieldNames){
+        PartialRow row = operation.getRow();
+        Schema colSchema = kuduTable.getSchema();
+
+        for (String colName : fieldNames) {
+            int colIdx = this.getColumnIndex(colSchema, colName);
+            if (colIdx != -1) {
+                Type colType = colSchema.getColumnByIndex(colIdx).getType();
+
+                switch (colType.getDataType()) {
+                    case BOOL:
+                        row.addBoolean(colIdx, record.getAsBoolean(colName));
+                        break;
+                    case FLOAT:
+                        row.addFloat(colIdx, record.getAsFloat(colName));
+                        break;
+                    case DOUBLE:
+                        row.addDouble(colIdx, record.getAsDouble(colName));
+                        break;
+                    case BINARY:
+                        row.addBinary(colIdx, record.getAsString(colName).getBytes());
+                        break;
+                    case INT8:
+                    case INT16:
+                        short temp = Short.parseShort(record.getAsInt(colName).toString());
```

Comment:
It's a bit inefficient to cast an int to string and then back to short. Instead you can just cast it like...

    short temp = (short) record.getAsInt(colName) 

If the value is out of range of a `short` it will just return as `-1`. If you try to parse the short as a string, it can cause an exception to be thrown (`java.lang.NumberFormatException`).